src/main/java/org/gridsuite/modification/server/NetworkModificationController.java

```diff
@@ -63,16 +62,24 @@ public ResponseEntity<Void> createModificationGroup(@RequestParam("groupUuid") U
-    public ResponseEntity<List<UUID>> updateModificationGroup(@Parameter(description = "group UUID") @PathVariable("groupUuid") UUID groupUuid,
+    public ResponseEntity<List<UUID>> updateModificationGroup(@Parameter(description = "updated group UUID, where modifications are pasted") @PathVariable("groupUuid") UUID groupUuid,
                                                               @Parameter(description = "kind of modification", required = true) @RequestParam(value = "action") GroupModificationAction action,
+                                                              @Parameter(description = "the network uuid", required = true) @RequestParam(value = "networkUuid") UUID networkUuid,
+                                                              @Parameter(description = "the report uuid", required = true) @RequestParam(value = "reportUuid") UUID reportUuid,
+                                                              @Parameter(description = "the reporter id", required = true) @RequestParam(value = "reporterId") UUID reporterId,
+                                                              @Parameter(description = "the variant id", required = true) @RequestParam(value = "variantId") String variantId,
                                                               @Parameter(description = "the modification Uuid to move before (MOVE option, empty means moving at the end)") @RequestParam(value = "before", required = false) UUID before,
-                                                              @RequestBody List<UUID> modificationsUuidList,
-                                                              @Nullable @RequestParam UUID originGroupUuid) {
+                                                              @Parameter(description = "origin group UUID, where modifications are copied or cut") @RequestParam(value = "originGroupUuid", required = false) UUID originGroupUuid,
+                                                              @Parameter(description = "destination node can be built (default is true)") @RequestParam(value = "build", required = false) Boolean build,
+                                                              @RequestBody List<UUID> modificationsUuidList) {
         switch (action) {
             case COPY:
-                return ResponseEntity.ok().body(networkModificationService.duplicateModifications(groupUuid, modificationsUuidList));
+                return ResponseEntity.ok().body(networkModificationService.duplicateModifications(groupUuid, networkUuid, reportUuid, reporterId, variantId, modificationsUuidList));
             case MOVE:
-                networkModificationService.moveModifications(groupUuid, originGroupUuid, before, modificationsUuidList);
+                // 2 params are optional
+                UUID sourceGroupUuid = originGroupUuid == null ? groupUuid : originGroupUuid;
+                boolean canBuildNode = build == null || build;
```

Comment:
it means the default behaviour is to build the node
I think it is better to add default value to the param like this :
**@RequestParam(value = "build", required = false, defaultValue = "true") Boolean build**

Reply:
yes, I will use defaultValue here